### DISCUSSION TITLE
Auth Proxy: Allow to set superadmin via a header

### DISCUSSION THREAD

**Discussion by UXabre** [grafana/grafana#44000] [General]
It seems that not many auth methods have the ability to describe a logged on user as being actually a superadmin; I think only ldap and OAuth seem to support it.

I propose to add another Header for auth proxy which can indicate that this user needs to be considered a superadmin. This way, one could utilize auth proxy to access the API, for instance, to create organizations.

A possible alternative, I think, is to support superadmin access tokens (although it's clunky compared to the auth proxy method; which, to me, is vital to have a perfect SSO solution)

**Comment by yeplaa** [comment]
As indicated in  https://github.com/grafana/grafana/issues/46104 , i have also the problem to install plugins from UI. It's not possible. 

In my use case, i have enabled AUTH_PROXY  and setted  *GF_USERS_AUTO_ASSIGN_ORG_ROLE* to 'Admin' value. On the UI, my user has a Admin role which is automatically associated but without some features.

Could you explain the difference between these 2 levels, Admin and Superadmin role? Maybe Superadmin could be associated to  *GF_USERS_AUTO_ASSIGN_ORG_ROLE* parameter.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82991. Feel free to continue the discussion around this there. Thank you!
